recaf-launcher/src/main/java/dev/xdark/recaf/Launcher.java

```diff
@@ -0,0 +1,399 @@
+      } else {
+        logger.error("Incompatible JVM version: {}", version);
+        logger.error("Please upgrade your installation to JDK 11+");
+        logger.error("Or install JDK that bundles JavaFX");
+        System.exit(1);
+      }
+    }
+
+    List<String> command = new LinkedList<>();
+    command.add(getJavaExecutable().toString());
+    command.add("-cp");
+    command.add(String.join(File.pathSeparator, classpath));
+    command.add(attributes.getValue("Main-Class"));
+    command.addAll(extraOptions);
+    logger.info("Starting new process: {}", command);
+    new ProcessBuilder()
+        .directory(jar.getParent().toFile())
+        .command(command)
+        .start();
+  }
+
+  private static void downloadDependencies(Path dir, String[] dependencies, List<String> classpath)
+      throws IOException {
+    for (String url : dependencies) {
+      Path path = dir.resolve(getCompactFileName(url));
+      if (!Files.exists(path)) {
+        logger.info("Downloading missing dependency: {} ----> {}", url, path);
+        byte[] content = IOUtils.toByteArray(new URL(url));
+        Files.write(path, content, StandardOpenOption.CREATE);
+      }
+      classpath.add(path.toString());
+    }
+  }
+
+  private static ReleaseInfo fetchReleaseInfo() throws IOException {
+    return GSON.fromJson(IOUtils.toString(new URL(API_URL),
+        StandardCharsets.UTF_8), ReleaseInfo.class);
+  }
+
+  private static boolean isOutdated(String current, String latest) {
+    String[] arrSelf = current.split("\\.");
+    String[] arrLatest = latest.split("\\.");
+    if (arrSelf.length == arrLatest.length) {
+      for (int i = 0; i < arrSelf.length; i++) {
+        int iSelf = Integer.parseInt(arrSelf[i]);
+        int iLatest = Integer.parseInt(arrLatest[i]);
+        if (iSelf != iLatest) {
+          return iSelf < iLatest;
+        }
+      }
+    }
+    return false;
+  }
+
+  private static Asset findReleaseAsset(List<Asset> assets) {
+    return assets.stream().filter(asset -> asset.getName().matches("recaf-*.*.*.jar")).findFirst()
+        .orElse(null);
+  }
+
+  private static Path getJavaHome() {
+    return Paths.get(System.getProperty("java.home"));
+  }
+
+  private static Path getJavaExecutable() {
+    Path javaHome = getJavaHome();
+    Path bin = javaHome.resolve("bin");
+    if (Platform.isOnWindows()) {
+      return bin.resolve("java.exe");
+    }
+    return bin.resolve("java");
+  }
+
+  private static Path getDirectory() {
+    Path directory = recafDirectory;
+    if (directory == null) {
+      try {
+        directory = Paths.get(BaseDirectories.get().configDir).resolve("Recaf");
+      } catch (Throwable t) {
+        if (!Platform.isOnWindows()) {
+          throw new IllegalStateException("Failed to initialize Recaf directory", t);
+        }
+        logger.error("Could not get base directory: ", t);
+        logger.error("Using %APPDATA%/Recaf as base directory");
+      }
+      if (directory == null) {
```

Comment:
May want to validate that we're on windows here. If we are, fetching APPDATA is 👍 
If we are not on windows, throw an illegal state indicating that base-directories failed on the current OS.

Reply:
Assuming that this directory may only be null on Windows, I did not add this check. Is it really necessary?

Reply:
I haven't read the base-dirs code for non-windows handling. So I only know of the powershell issue we had before. 

Not really necessary, just thinking about being safe.